addons/addon-base-raas/packages/base-raas-services/lib/data-egress/__test__/data-egress-service.test.js

```diff
@@ -580,6 +580,294 @@ describe('DataEgressService', () => {
+      const mockRequestContext = { principalIdentifier: { uid: 'test-createdBy' } };
+      await expect(dataEgressService.notifySNS(mockRequestContext, 'workspaceId')).rejects.toThrow(
+        // It is better to check using boom.code instead of just the actual string, unless
+        // there are a few errors with the exact same boom code but different messages.
+        // Note: if you encounter a case where a service is throwing exceptions with the
+        // same code but different messages (to address different scenarios), you might
+        // want to suggest to the service author to use different codes.
+        expect.objectContaining({ boom: true, code: 'forbidden', safe: true }),
+      );
+    });
+
+    it('should notify SNS', async () => {
+      dataEgressService.audit = jest.fn();
+      dataEgressService._settings = {
+        get: settingName => {
+          if (settingName === 'egressNotificationBucketName') {
+            return 'test-egressNotificationBucketName';
+          }
+          if (settingName === 'enableEgressStore') {
+            return 'true';
+          }
+          if (settingName === 'egressNotificationSnsTopicArn') {
+            return 'test-egressNotificationSnsTopicArn';
+          }
+          return undefined;
+        },
+      };
+
+      const mockEgressStoreInfo = {
+        id: 'id',
+        egressStoreName: 'egressStoreName',
+        createdAt: 'createdAt',
+        createdBy: 'createdBy',
+        workspaceId: 'workspaceId',
+        projectId: 'projectId',
+        s3BucketName: 's3BucketName',
+        s3BucketPath: 's3BucketPath',
+        status: 'status',
+        updatedBy: 'updatedBy',
+        updatedAt: 'updatedAt',
+        ver: 'ver',
+        isAbleToSubmitEgressRequest: true,
+      };
+      dbService.table.scan.mockResolvedValue([mockEgressStoreInfo]);
+      const mockRequestContext = { principalIdentifier: { uid: 'createdBy' } };
+      dataEgressService.lockAndUpdate = jest.fn();
+      dataEgressService.publishMessage = jest.fn();
+
+      await dataEgressService.notifySNS(mockRequestContext, 'workspaceId');
+      expect(dataEgressService.audit).toHaveBeenCalledTimes(1);
+      expect(dataEgressService.audit).toHaveBeenCalledWith(
+        { principalIdentifier: { uid: 'createdBy' } },
+        {
+          action: 'trigger-egress-notification-process',
+          body: {
+            created_at: 'createdAt',
+            created_by: 'createdBy',
+            egress_store_id: 'id',
+            egress_store_name: 'egressStoreName',
+            egressStoreObjectListLocation:
+              'arn:aws:s3:::test-egressNotificationBucketName/id/egressStoreName-verNaN.json',
+            id: expect.anything(),
+            project_id: 'projectId',
+            s3_bucketname: 's3BucketName',
+            s3_bucketpath: 's3BucketPath',
+            status: 'PENDING',
+            updated_at: expect.anything(),
+            updated_by: 'createdBy',
+            ver: NaN,
```

Comment:
Why NaN?

Reply:
fixed in commit: 18639d639b979c9054fa5d24fa94b3a0bdd98f11